### DISCUSSION TITLE
Need help with aiohttp error

### DISCUSSION THREAD

**Discussion by pranpraveen** [aio-libs/aiohttp#7276] [Q&A]
Hello Gents,

Any idea why I am getting this error. Appreciate any help in trouble shooting this.

Building wheels for collected packages: aiohttp
  Building wheel for aiohttp (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for aiohttp (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [127 lines of output]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\abc.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\base_protocol.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client_exceptions.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client_proto.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client_reqrep.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client_ws.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\connector.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\cookiejar.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\formdata.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\hdrs.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\helpers.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\http.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\http_exceptions.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\http_parser.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\http_websocket.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\http_writer.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\locks.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\log.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\multipart.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\payload.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\payload_streamer.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\pytest_plugin.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\resolver.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\streams.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\tcp_helpers.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\test_utils.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\tracing.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\typedefs.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_app.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_exceptions.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_fileresponse.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_log.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_middlewares.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_protocol.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_request.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_response.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_routedef.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_runner.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_server.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_urldispatcher.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\web_ws.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\worker.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\__init__.py -> build\lib.win-amd64-cpython-311\aiohttp
      running egg_info
      writing aiohttp.egg-info\PKG-INFO
      writing dependency_links to aiohttp.egg-info\dependency_links.txt
      writing requirements to aiohttp.egg-info\requires.txt
      writing top-level names to aiohttp.egg-info\top_level.txt
      reading manifest file 'aiohttp.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching 'aiohttp' anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.pyd' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '*.lib' found anywhere in distribution
      warning: no previously-included files matching '*.dll' found anywhere in distribution
      warning: no previously-included files matching '*.a' found anywhere in distribution
      warning: no previously-included files matching '*.obj' found anywhere in distribution
      warning: no previously-included files found matching 'aiohttp\*.html'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE.txt'
      writing manifest file 'aiohttp.egg-info\SOURCES.txt'
      copying aiohttp\_cparser.pxd -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_find_header.c -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_find_header.h -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_find_header.pxd -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_headers.pxi -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_helpers.c -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_helpers.pyi -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_helpers.pyx -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_http_parser.c -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_http_parser.pyx -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_http_writer.c -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_http_writer.pyx -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_websocket.c -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_websocket.pyx -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\py.typed -> build\lib.win-amd64-cpython-311\aiohttp
      creating build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_cparser.pxd.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_find_header.pxd.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_helpers.pyi.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_helpers.pyx.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_http_parser.pyx.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_http_writer.pyx.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\_websocket.pyx.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      copying aiohttp\.hash\hdrs.py.hash -> build\lib.win-amd64-cpython-311\aiohttp\.hash
      running build_ext
      building 'aiohttp._websocket' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\aiohttp
      "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\<ID>\AppData\Local\Programs\Python\Python310\include -IC:\Users\<ID>\AppData\Local\Programs\Python\Python310\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" /Tcaiohttp/_websocket.c /Fobuild\temp.win-amd64-cpython-311\Release\aiohttp/_websocket.obj
      _websocket.c
      aiohttp/_websocket.c(1716): warning C4047: 'function': 'PyObject *' differs in levels of indirection from 'int'
      aiohttp/_websocket.c(1716): warning C4024: 'PyCode_New': different types for formal and actual parameter 14
      aiohttp/_websocket.c(1716): warning C4047: 'function': 'int' differs in levels of indirection from 'PyObject *'
      aiohttp/_websocket.c(1716): warning C4024: 'PyCode_New': different types for formal and actual parameter 15
      aiohttp/_websocket.c(1716): error C2198: 'PyCode_New': too few arguments for call
      aiohttp/_websocket.c(1944): error C2027: use of undefined type '_frame'
      C:\Users\<ID>\AppData\Local\Programs\Python\Python310\include\pytypedefs.h(22): note: see declaration of '_frame'
      aiohttp/_websocket.c(1944): error C2037: left of 'f_localsplus' specifies undefined struct/union '_frame'
      aiohttp/_websocket.c(1944): error C2056: illegal expression
      aiohttp/_websocket.c(1944): error C2466: cannot allocate an array of constant size 0
      aiohttp/_websocket.c(1944): warning C4034: sizeof returns 0
      aiohttp/_websocket.c(2306): error C2039: 'recursion_depth': is not a member of '_ts'
      C:\Users\<ID>\AppData\Local\Programs\Python\Python310\include\cpython/pystate.h(82): note: see declaration of '_ts'
      aiohttp/_websocket.c(2308): error C2039: 'recursion_depth': is not a member of '_ts'
      C:\Users\<ID>\AppData\Local\Programs\Python\Python310\include\cpython/pystate.h(82): note: see declaration of '_ts'
      aiohttp/_websocket.c(2335): error C2065: 'CO_NOFREE': undeclared identifier
      aiohttp/_websocket.c(2760): warning C4047: 'function': 'PyObject *' differs in levels of indirection from 'int'
      aiohttp/_websocket.c(2744): warning C4024: 'PyCode_New': different types for formal and actual parameter 14
      aiohttp/_websocket.c(2760): warning C4047: 'function': 'int' differs in levels of indirection from 'PyObject *'
      aiohttp/_websocket.c(2744): warning C4024: 'PyCode_New': different types for formal and actual parameter 15
      aiohttp/_websocket.c(2744): error C2198: 'PyCode_New': too few arguments for call
      aiohttp/_websocket.c(2791): error C2037: left of 'f_lineno' specifies undefined struct/union '_frame'
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for aiohttp
Failed to build aiohttp
ERROR: Could not build wheels for aiohttp, which is required to install pyproject.toml-based projects

**Comment by Dreamsorcerer** [comment]
No idea, but you could start by including relevant information (what command did you run, version of pip/Python/aiohttp, architecture...).
